clang/lib/Sema/SemaExprMember.cpp

```diff
@@ -1655,8 +1655,10 @@ static ExprResult LookupMemberExpr(Sema &S, LookupResult &R,
     // We disallow element access for ext_vector_type bool.  There is no way to
     // materialize a reference to a vector element as a pointer (each element is
     // one bit in the vector).
+    assert(MemberName.isIdentifier() &&
+           "Ext vector component name not an identifier!");
     S.Diag(R.getNameLoc(), diag::err_ext_vector_component_name_illegal)
-        << MemberName
+        << MemberName.getAsIdentifierInfo()->getName()
```

Comment:
Did you try to remove the quotes in DiagnositicSemaKind.td instead?

Reply:
The reason I did this instead of that was cause in `CheckExtVectorComponent`, a `StringRef` is passed to the diagnostics handler, which doesn't add quotes around the name, hence why the quotes are in the error message in the first place.   

Removing the quotes in `DiagnosticsSemaKind.td` causes the tests in `/Sema/ext_vector_components.c` to fail.

The real todo would be to implement modifications for `ak_std_string`. 

Let me know what you think, also thank you for taking the time to look at this.

Reply:
I would rather we remove the quotes from the diagnostic, and re-add them in the string-ref being passed (OR, better, refactor that part to actually send the component rather than the string ref).  Having to do this work to get the identifier that should be handled by the diagnostics engine isn't correct IMO.

That is, remove the quotes from the diag, and fix `CheckExtVectorComponent`, not make this site worse.

Reply:
Fair enough. I'll put this back into draft and get to work on that.

Thanks!